plugins/java-decompiler/engine/src/org/jetbrains/java/decompiler/struct/StructClass.java

```diff
@@ -162,6 +164,24 @@ public ConstantPool getPool() {
     return fields;
   }
 
+  // XXX slow, but no lookup table available
+  public boolean containsFieldWithShortName(String shortName){
+    for(StructField f: fields) {
+      if(shortName.equals(f.getName())) {
```

Comment:
The new test case 'pkg/TestSwitchOnEnum' illustrates the NPE. This was closed with simple NPE guard in the commit  bcc6a3b
Check HeadsUp on commit message.